README.md

```diff
@@ -1 +1,649 @@
+| `sig`  | `sign(requestorPK, sha3_256(responderDid + outOfBandPin))` | Signature of challenge hash | Yes      |
+
+```javascript
+{
+  "did": requestorActualDid,
+  "sig": signedHash
+}
+```
+
+#### 3.4.2.3 Direct UCAN Challenge
+
+If UCAN auth is required by the Responder, the Requestor MUST provide a UCAN. This is the same strategy as the one used by the Responder in [§3.3](#33-responder-es tablishes-point-to-point-session): the UCAN MUST be encrypted with the session key and the IV from the enclosing payload, MUST be given in a raw format, and MUST be inline (without a JSON object wrapper or similar).
+
+The UCAN MUST be issued (`iss`) by the Requestor's DID (not the temporary DID), and its audience (`aud`) MUST be the Responder's DID. The `att` field MUST be set to an empty array (i.e. it MUST NOT delegate any capabilities). The `prf` array MUST fulfill the capabilities set by the Responder.
+
+```
+              UCAN Auth
+
+┌──────────────AES-GCM────────────┐
+│                                 │
+│  ┌────────────UCAN───────────┐  │
+│  │                           │  │
+│  │  iss: RequestorActualDid  │  │
+│  │  aud: ResponderActualDid  │  │
+│  │  fct: nextReqECDH         │  │
+│  │  att: []                  │  │
+│  │  prf: ...                 │  │
+│  │                           │  │
+│  └───────────────────────────┘  │
+│                                 │
+└─────────────────────────────────┘
+```
+
+# 3.5 Responder Acknowledgment
+
+The Responder MUST respond with an acknowledgment that the challenge in [§3.4](#34-requestor-challenge) was accepted.
+
+```
+Requestor                  Responder
+    ⋮                          ⋮
+    │           ACK            │ (5)
+    │◄─────────────────────────┤
+    │                          │
+```
+
+### 3.5.1 Payload
+
+| Field  | Value                             | Description                                                    | Required |
+| ------ | --------------------------------- | -------------------------------------------------------------- | -------- |
+| `awv`  | `"0.1.0"`                         | AWAKE message version                                          | Yes      |
+| `type` | `"awake/msg"`                     | Generic AWAKE message type                                     | Yes      |
+| `id`   | `sha3_256(resEcdhPk + reqEcdhPk)` | Message ID                                                     | Yes      |
+| `iv`   |                                   | Initialization vector for the encrypted payload                | Yes      |
+| `msg`  |                                   | Fulfilled challenge payload encrypted with Step 4 ECDH AES-key | Yes      |
+
+#### 3.5.1.1 Encrypted Message
+
+The encrypted message payload MUST include an `awake/ack` field, with a value of the requestor's long-term DID. This payload MAY contain additional fields. This is often useful if dovetailing the ACK with the first message of a session using the 
```

Comment:
Did this last sentence get cut off?

Reply:
![](https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Fwww.memesmonkey.com%2Fimages%2Fmemesmonkey%2F59%2F5976e7cfca942a1a9f8b747600dea081.jpeg&f=1&nofb=1)